src/cpp/proxy/app.cpp

```diff
@@ -162,21 +165,149 @@ static CommandLineParseResult parseCommandLine(QCommandLineParser *parser, ArgsD
 	return CommandLineOk;
 }
 
+class EngineWorker : public QObject
```

Comment:
can we refrain from adding QObject to our new code logic as we are moving away from this approach all together

Reply:
fair question! here it is still needed, due to `QMetaObject::invokeMethod`.

Reply:
due to QMetaObject::invokeMethod we need to include `Q_OBJECT` macro but I do not think we need the `QObject` as a parent class